src/snowflake/snowpark/dataframe.py

```diff
@@ -643,6 +644,9 @@ def __init__(
 
         self._alias: Optional[str] = None
 
+        if context._debug_eager_schema_validation:
+            self._plan.attributes
```

Comment:
nit: I think we can leave a comment on calling `attribute` eagerly evaluates the plan.